### DISCUSSION TITLE
i3 configuration file and recursive variable parsing

### DISCUSSION THREAD

**Discussion by DarkPoulet** [i3/i3#5099] [Q&A]
Hello. 

In the [documentation](https://i3wm.org/docs/userguide.html#variables), it is written "Variables are directly replaced in the file when parsing. Variables expansion is not recursive so it is not possible to define a variable with a value containing another variable. There is no fancy handling and there are absolutely no plans to change this." 

I am wondering why and did not find an answer in this FAQ. 
Is there any particular difficulty or inconvenience in use in implementing this functionality? 

It would be very helpful in order to better organize themes or even long / repetitive commands.

Thank you.

**Comment by Thaodan** [comment]
That's the only reference I could find:
https://github.com/i3/i3/issues/5062#issuecomment-1198244318

Someone has gone ahead and implement the feature:
https://github.com/Cubified/i3-gaps-recursive#recursive-variables
